---
title: "Custom genes for 10xscRNA"
output:
  html_document:
    df_print: paged
    toc: true
    toc_float: true
---

## Intro Single cell(10x) 

Fluorescent protein of tags are commonly used to label specific cell populations. When these are subjected to single cell sequencing, they can be used to confirm that if specific genes are co-expressed, they aren't a just coming from a doublet or to even calculate doublet rate. 
Instructions for adding custom genes can be found at the [10X genomics] (https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/advanced/references).
I'll give step by step instruction to do so. 

## Setup
1.Download the reference files from [10X](https://support.10xgenomics.com/single-cell-gene-expression/software/downloads/latest) and unzip into `refdata-cellranger-mm10-3.0.0/`. Make sure to read the build instructions.

2.Make a new folder `reference`. 

3.`cat > temp_genome.fa`. Edit in a text editor. 
I made a new chromosome for each gene, but you can just make one locus. This would require you to create a locus with the appropriate sequence with `NNN` in between them. You may also just consider adding the genes to chromosome 1, where you replace the `NNN` with your sequence. Just make sure to that you track the nucleotide number correctly. I found it eaiser to just have a new chromosome for each sequence. 

You can get specific sequences for many common features from [Addgene](https://www.addgene.org/tools/reference/plasmid-features/).Make sure it is CAPITALIZED. They don't need to be in FASTA format and can just be in a long string. Remeber that the FASTA format give an additional space at the end of each line so your character count may be off if you do it. 

`>eGFP`
`GTGAGCAAGGGCGAGGAGCTGTTCACCGGGGTGGTGCCCATCCTGGTCGAGCTGGACGGC
GACGTAAACGGCCACAAGTTCAGCGTGTCCGGCGAGGGCGAGGGCGATGCCACCTACGGC
AAGCTGACCCTGAAGTTCATCTGCACCACCGGCAAGCTGCCCGTGCCCTGGCCCACCCTC
GTGACCACCCTGACCTACGGCGTGCAGTGCTTCAGCCGCTACCCCGACCACATGAAGCAG
CACGACTTCTTCAAGTCCGCCATGCCCGAAGGCTACGTCCAGGAGCGCACCATCTTCTTC
AAGGACGACGGCAACTACAAGACCCGCGCCGAGGTGAAGTTCGAGGGCGACACCCTGGTG
AACCGCATCGAGCTGAAGGGCATCGACTTCAAGGAGGACGGCAACATCCTGGGGCACAAG
CTGGAGTACAACTACAACAGCCACAACGTCTATATCATGGCCGACAAGCAGAAGAACGGC
ATCAAGGTGAACTTCAAGATCCGCCACAACATCGAGGACGGCAGCGTGCAGCTCGCCGAC
CACTACCAGCAGAACACCCCCATCGGCGACGGCCCCGTGCTGCTGCCCGACAACCACTAC
CTGAGCACCCAGTCCGCCCTGAGCAAAGACCCCAACGAGAAGCGCGATCACATGGTCCTG
CTGGAGTTCGTGACCGCCGCCGGGATCACTCTCGGCATGGACGAGCTGTACAAG`
`>eGFPSV40`
`ATGGTGAGCAAGGGCGAGGAGCTGTTCACCGGGGTGGTGCCCATCCTGGTCGAGCTGGAC
GGCGACGTAAACGGCCACAAGTTCAGCGTGTCCGGCGAGGGCGAGGGCGATGCCACCTAC
GGCAAGCTGACCCTGAAGTTCATCTGCACCACCGGCAAGCTGCCCGTGCCCTGGCCCACC
CTCGTGACCACCCTGACCTACGGCGTGCAGTGCTTCAGCCGCTACCCCGACCACATGAAG
CAGCACGACTTCTTCAAGTCCGCCATGCCCGAAGGCTACGTCCAGGAGCGCACCATCTTC
TTCAAGGACGACGGCAACTACAAGACCCGCGCCGAGGTGAAGTTCGAGGGCGACACCCTG
GTGAACCGCATCGAGCTGAAGGGCATCGACTTCAAGGAGGACGGCAACATCCTGGGGCAC
AAGCTGGAGTACAACTACAACAGCCACAACGTCTATATCATGGCCGACAAGCAGAAGAAC
GGCATCAAGGTGAACTTCAAGATCCGCCACAACATCGAGGACGGCAGCGTGCAGCTCGCC
GACCACTACCAGCAGAACACCCCCATCGGCGACGGCCCCGTGCTGCTGCCCGACAACCAC
TACCTGAGCACCCAGTCCGCCCTGAGCAAAGACCCCAACGAGAAGCGCGATCACATGGTC
CTGCTGGAGTTCGTGACCGCCGCCGGGATCACTCTCGGCATGGACGAGCTGTACAAGTAA
AGCGGCCGGCCGCGACTCTAGATCATAATCAGCCATACCACATTTGTAGAGGTTTTACTT
GCTTTAAAAAACCTCCCACACCTCCCCCTGAACCTGAAACATAAAATGAATGCAATTGTT
GTTGTTAACTTGTTTATTGCAGCTTATAATGGTTACAAATAAAGCAATAGCATCACAAAT
TTCACAAATAAAGCATTTTTTTCACTGCATTCTAGTTGTGGTTTGTCCAAACTCATCAAT
GTATCTTA`
> The tdTomatoWPRE construct was from the Ai9 mouse and the vectors information was obtained from [Adgene](http://www.addgene.org/browse/sequence/115994/)
`>tdTomatoWPRE`
`ATGGTGAGCAAGGGCGAGGAGGTCATCAAAGAGTTCATGCGCTTCAAGGTGCGCATGGAGGGCTCCATGAACGGCCACGAGTTCGAGATCGAGGGCGAGGGCGAGGGCCGCCCCTACGAGGGCACCCAGACCGCCAAGCTGAAGGTGACCAAGGGCGGCCCCCTGCCCTTCGCCTGGGACATCCTGTCCCCCCAGTTCATGTACGGCTCCAAGGCGTACGTGAAGCACCCCGCCGACATCCCCGATTACAAGAAGCTGTCCTTCCCCGAGGGCTTCAAGTGGGAGCGCGTGATGAACTTCGAGGACGGCGGTCTGGTGACCGTGACCCAGGACTCCTCCCTGCAGGACGGCACGCTGATCTACAAGGTGAAGATGCGCGGCACCAACTTCCCCCCCGACGGCCCCGTAATGCAGAAGAAGACCATGGGCTGGGAGGCCTCCACCGAGCGCCTGTACCCCCGCGACGGCGTGCTGAAGGGCGAGATCCACCAGGCCCTGAAGCTGAAGGACGGCGGCCACTACCTGGTGGAGTTCAAGACCATCTACATGGCCAAGAAGCCCGTGCAACTGCCCGGCTACTACTACGTGGACACCAAGCTGGACATCACCTCCCACAACGAGGACTACACCATCGTGGAACAGTACGAGCGCTCCGAGGGCCGCCACCACCTGTTCCTGGGGCATGGCACCGGCAGCACCGGCAGCGGCAGCTCCGGCACCGCCTCCTCCGAGGACAACAACATGGCCGTCATCAAAGAGTTCATGCGCTTCAAGGTGCGCATGGAGGGCTCCATGAACGGCCACGAGTTCGAGATCGAGGGCGAGGGCGAGGGCCGCCCCTACGAGGGCACCCAGACCGCCAAGCTGAAGGTGACCAAGGGCGGCCCCCTGCCCTTCGCCTGGGACATCCTGTCCCCCCAGTTCATGTACGGCTCCAAGGCGTACGTGAAGCACCCCGCCGACATCCCCGATTACAAGAAGCTGTCCTTCCCCGAGGGCTTCAAGTGGGAGCGCGTGATGAACTTCGAGGACGGCGGTCTGGTGACCGTGACCCAGGACTCCTCCCTGCAGGACGGCACGCTGATCTACAAGGTGAAGATGCGCGGCACCAACTTCCCCCCCGACGGCCCCGTAATGCAGAAGAAGACCATGGGCTGGGAGGCCTCCACCGAGCGCCTGTACCCCCGCGACGGCGTGCTGAAGGGCGAGATCCACCAGGCCCTGAAGCTGAAGGACGGCGGCCACTACCTGGTGGAGTTCAAGACCATCTACATGGCCAAGAAGCCCGTGCAACTGCCCGGCTACTACTACGTGGACACCAAGCTGGACATCACCTCCCACAACGAGGACTACACCATCGTGGAACAGTACGAGCGCTCCGAGGGCCGCCACCACCTGTTCCTGTACGGCATGGACGAGCTGTACAAGTAAGAATTGTGTTGCACTTAACGCGTACAAGGCCGGCCCTGCAGGAATTCGATATCAAGCTTATCGATAATCAACCTCTGGATTACAAAATTTGTGAAAGATTGACTGGTATTCTTAACTATGTTGCTCCTTTTACGCTATGTGGATACGCTGCTTTAATGCCTTTGTATCATGCTATTGCTTCCCGTATGGCTTTCATTTTCTCCTCCTTGTATAAATCCTGGTTGCTGTCTCTTTATGAGGAGTTGTGGCCCGTTGTCAGGCAACGTGGCGTGGTGTGCACTGTGTTTGCTGACGCAACCCCCACTGGTTGGGGCATTGCCACCACCTGTCAGCTCCTTTCCGGGACTTTCGCTTTCCCCCTCCCTATTGCCACGGCGGAACTCATCGCCGCCTGCCTTGCCCGCTGCTGGACAGGGGCTCGGCTGTTGGGCACTGACAATTCCGTGGTGTTGTCGGGGAAATCATCGTCCTTTCCTTGGCTGCTCGCCTGTGTTGCCACCTGGATTCTGCGCGGGACGTCCTTCTGCTACGTCCCTTCGGCCCTCAATCCAGCGGACCTTCCTTCCCGCGGCCTGCTGCCGGCTCTGCGGCCTCTTCCGCGTCTTCGCCTTCGCCCTCAGACGAGTCGGATCTCCCTTTGGGCCGCCTCCCCGCATCGATACCGTCGACCTCGACCTCGA`

> WPRE 
taatcaacctctggattacaaaatttgtgaaagattgactggtattcttaactatgttgctccttttacgctatgtggatacgctgctttaatgcctttgtatcatgctattgcttcccgtatggctttcattttctcctccttgtataaatcctggttgctgtctctttatgaggagttgtggcccgttgtcaggcaacgtggcgtggtgtgcactgtgtttgctgacgcaacccccactggttggggcattgccaccacctgtcagctcctttccgggactttcgctttccccctccctattgccacggcggaactcatcgccgcctgccttgcccgctgctggacaggggctcggctgttgggcactgacaattccgtggtgttgtcggggaaatcatcgtcctttccttggctgctcgcctgtgttgccacctggattctgcgcgggacgtccttctgctacgtcc

#Have not tried wpre by itself. [ ] Check WPRE mapping. 

>WPREBGHPolyA 
TAATCAACCTCTGGATTACAAAATTTGTGAAAGATTGACTGGTATTCTTAACTATGTTGCTCCTTTTACGCTATGTGGATACGCTGCTTTAATGCCTTTGTATCATGCTATTGCTTCCCGTATGGCTTTCATTTTCTCCTCCTTGTATAAATCCTGGTTGCTGTCTCTTTATGAGGAGTTGTGGCCCGTTGTCAGGCAACGTGGCGTGGTGTGCACTGTGTTTGCTGACGCAACCCCCACTGGTTGGGGCATTGCCACCACCTGTCAGCTCCTTTCCGGGACTTTCGCTTTCCCCCTCCCTATTGCCACGGCGGAACTCATCGCCGCCTGCCTTGCCCGCTGCTGGACAGGGGCTCGGCTGTTGGGCACTGACAATTCCGTGGTGTTGTCGGGGAAATCATCGTCCTTTCCTTGGCTGCTCGCCTGTGTTGCCACCTGGATTCTGCGCGGGACGTCCTTCTGCTACGTCCCTTCGGCCCTCAATCCAGCGGACCTTCCTTCCCGCGGCCTGCTGCCGGCTCTGCGGCCTCTTCCGCGTCTTCGCCTTCGCCCTCAGACGAGTCGGATCTCCCTTTGGGCCGCCTCCCCGCATCGATACCGTCGACCTCGACCTCGACTGTGCCTTCTAGTTGCCAGCCATCTGTTGTTTGCCCCTCCCCCGTGCCTTCCTTGACCCTGGAAGGTGCCACTCCCACTGTCCTTTCCTAATAAAATGAGGAAATTGCATCGCATTGTCTGAGTAGGTGTCATTCTATTCTGGGGGGTGGGGTGGGGCAGGACAGCAAGGGGGAGGATTGGGAAGACAATGGCAGGCATGCTGGGG
#This contains the long SV40PolyA with the 3' found from the mapped reads. 
>SV40_polyA
GATCATAATCAGCCATACCACATTTGTAGAGGTTTTACTT
GCTTTAAAAAACCTCCCACACCTCCCCCTGAACCTGAAACATAAAATGAATGCAATTGTT
GTTGTTAACTTGTTTATTGCAGCTTATAATGGTTACAAATAAAGCAATAGCATCACAAAT
TTCACAAATAAAGCATTTTTTTCACTGCATTCTAGTTGTGGTTTGTCCAAACTCATCAAT
GTATCTTA

#The one above did not work. 
I would like to also map the Pax7-IRES-CreERT2. However this is a little complicated. The CreERT2 is ~2015 bp long and 10X is 3' biased. When you use the ERT2 sequence, most of the reads are thrown off because it has ~100% homology to Esr1(estrogen receptor 1). So you have to find the bits of the 3'UTR. 

>CreERT2 
ATGGCCAATTTACTGACCGTACACCAAAATTTGCCTGCATTACCGGTCGATGCAACGAGT
GATGAGGTTCGCAAGAACCTGATGGACATGTTCAGGGATCGCCAGGCGTTTTCTGAGCAT
ACCTGGAAAATGCTTCTGTCCGTTTGCCGGTCGTGGGCGGCATGGTGCAAGTTGAATAAC
CGGAAATGGTTTCCCGCAGAACCTGAAGATGTTCGCGATTATCTTCTATATCTTCAGGCG
CGCGGTCTGGCAGTAAAAACTATCCAGCAACATTTGGGCCAGCTAAACATGCTTCATCGT
CGGTCCGGGCTGCCACGACCAAGTGACAGCAATGCTGTTTCACTGGTTATGCGGCGGATC
CGAAAAGAAAACGTTGATGCCGGTGAACGTGCAAAACAGGCTCTAGCGTTCGAACGCACT
GATTTCGACCAGGTTCGTTCACTCATGGAAAATAGCGATCGCTGCCAGGATATACGTAAT
CTGGCATTTCTGGGGATTGCTTATAACACCCTGTTACGTATAGCCGAAATTGCCAGGATC
AGGGTTAAAGATATCTCACGTACTGACGGTGGGAGAATGTTAATCCATATTGGCAGAACG
AAAACGCTGGTTAGCACCGCAGGTGTAGAGAAGGCACTTAGCCTGGGGGTAACTAAACTG
GTCGAGCGATGGATTTCCGTCTCTGGTGTAGCTGATGATCCGAATAACTACCTGTTTTGC
CGGGTCAGAAAAAATGGTGTTGCCGCGCCATCTGCCACCAGCCAGCTATCAACTCGCGCC
CTGGAAGGGATTTTTGAAGCAACTCATCGATTGATTTACGGCGCTAAGGATGACTCTGGT
CAGAGATACCTGGCCTGGTCTGGACACAGTGCCCGTGTCGGAGCCGCGCGAGATATGGCC
CGCGCTGGAGTTTCAATACCGGAGATCATGCAAGCTGGTGGCTGGACCAATGTAAATATT
GTCATGAACTATATCCGTAACCTGGATAGTGAAACAGGGGCAATGGTGCGCCTGCTGGAA
GATGGCGATCTCGAGCCATCTGCTGGAGACATGAGAGCTGCCAACCTTTGGCCAAGCCCG
CTCATGATCAAACGCTCTAAGAAGAACAGCCTGGCCTTGTCCCTGACGGCCGACCAGATG
GTCAGTGCCTTGTTGGATGCTGAGCCCCCCATACTCTATTCCGAGTATGATCCTACCAGA
CCCTTCAGTGAAGCTTCGATGATGGGCTTACTGACCAACCTGGCAGACAGGGAGCTGGTT
CACATGATCAACTGGGCGAAGAGGGTGCCAGGCTTTGTGGATTTGACCCTCCATGATCAG
GTCCACCTTCTAGAATGTGCCTGGCTAGAGATCCTGATGATTGGTCTCGTCTGGCGCTCC
ATGGAGCACCCAGTGAAGCTACTGTTTGCTCCTAACTTGCTCTTGGACAGGAACCAGGGA
AAATGTGTAGAGGGCATGGTGGAGATCTTCGACATGCTGCTGGCTACATCATCTCGGTTC
CGCATGATGAATCTGCAGGGAGAGGAGTTTGTGTGCCTCAAATCTATTATTTTGCTTAAT
TCTGGAGTGTACACATTTCTGTCCAGCACCCTGAAGTCTCTGGAAGAGAAGGACCATATC
CACCGAGTCCTGGACAAGATCACAGACACTTTGATCCACCTGATGGCCAAGGCAGGCCTG
ACCCTGCAGCAGCAGCACCAGCGGCTGGCCCAGCTCCTCCTCATCCTCTCCCACATCAGG
CACATGAGTAACAAAGGCATGGAGCATCTGTACAGCATGAAGTGCAAGAACGTGGTGCCC
CTCTATGACCTGCTGCTGGAGGCGGCGGACGCCCACCGCCTACATGCGCCCACTAGCCGT
GGAGGGGCATCCGTGGAGGAGACGGACCAAAGCCACTTGGCCACTGCGGGCTCTACTTCA
TCGCATTCCTTGCAAAAGTATTACATCACGGGGGAGGCAGAGGGTTTCCCTGCCACAGCT
TG 

This below does not work because its has ~100% homology to Esr1(estrogen receptor1)
>ErT2
GCTGGAGACATGAGAGCTGCCAACCTTTGGCCAAGCCCGCTCATGATCAAACGCTCTAAG
AAGAACAGCCTGGCCTTGTCCCTGACGGCCGACCAGATGGTCAGTGCCTTGTTGGATGCT
GAGCCCCCCATACTCTATTCCGAGTATGATCCTACCAGACCCTTCAGTGAAGCTTCGATG
ATGGGCTTACTGACCAACCTGGCAGACAGGGAGCTGGTTCACATGATCAACTGGGCGAAG
AGGGTGCCAGGCTTTGTGGATTTGACCCTCCATGATCAGGTCCACCTTCTAGAATGTGCC
TGGCTAGAGATCCTGATGATTGGTCTCGTCTGGCGCTCCATGGAGCACCCAGTGAAGCTA
CTGTTTGCTCCTAACTTGCTCTTGGACAGGAACCAGGGAAAATGTGTAGAGGGCATGGTG
GAGATCTTCGACATGCTGCTGGCTACATCATCTCGGTTCCGCATGATGAATCTGCAGGGA
GAGGAGTTTGTGTGCCTCAAATCTATTATTTTGCTTAATTCTGGAGTGTACACATTTCTG
TCCAGCACCCTGAAGTCTCTGGAAGAGAAGGACCATATCCACCGAGTCCTGGACAAGATC
ACAGACACTTTGATCCACCTGATGGCCAAGGCAGGCCTGACCCTGCAGCAGCAGCACCAG
CGGCTGGCCCAGCTCCTCCTCATCCTCTCCCACATCAGGCACATGAGTAACAAAGGCATG
GAGCATCTGTACAGCATGAAGTGCAAGAACGTGGTGCCCCTCTATGACCTGCTGCTGGAG
GCGGCGGACGCCCACCGCCTACATGCGCCCACTAGCCGTGGAGGGGCATCCGTGGAGGAG
ACGGACCAAAGCCACTTGGCCACTGCGGGCTCTACTTCATCGCATTCCTTGCAAAAGTAT
TACATCACGGGGGAGGCAGAGGGTTTCCCTG

Got the map from Gab for the Pax7-IRES-CreERT2.
![Pax7CreERT2 Map](./images/Pax7CreERT2Map.png). 
So this is the whole IRES-CREERT2-FRT-Pax7exon9incomplete-attB2. I will call it ICEPax7
>ICEPax7
GCCCCTCTCCCTCCCCCCCCCCTAACGTTACTGGCCGAAGCCGCTTGGAATAAGGCCGGTGTGCGTTTGTCTATATGTTATTTTCCACCATATTGCCGTCTTTTGGCAATGTGAGGGCCCGGAAACCTGGCCCTGTCTTCTTGACGAGCATTCCTAGGGGTCTTTCCCCTCTCGCCAAAGGAATGCAAGGTCTGTTGAATGTCGTGAAGGAAGCAGTTCCTCTGGAAGCTTCTTGAAGACAAACAACGTCTGTAGCGACCCTTTGCAGGCAGCGGAACCCCCCACCTGGCGACAGGTGCCTCTGCGGCCAAAAGCCACGTGTATAAGATACACCTGCAAAGGCGGCACAACCCCAGTGCCACGTTGTGAGTTGGATAGTTGTGGAAAGAGTCAAATGGCTCTCCTCAAGCGTATTCAACAAGGGGCTGAAGGATGCCCAGAAGGTACCCCATTGTATGGGATCTGATCTGGGGCCTCGGTACACATGCTTTACATGTGTTTAGTCGAGGTTAAAAAAACGTCTAGGCCCCCCGAACCACGGGGACGTGGTTTTCCTTTGAAAAACACGATGATAATATGGCCACAACGAATTCCACCATGTCCAATTTACTGACCGTACACCAAAATTTGCCTGCATTACCGGTCGATGCAACGAGTGATGAGGTTCGCAAGAACCTGATGGACATGTTCAGGGATCGCCAGGCGTTTTCTGAGCATACCTGGAAAATGCTTCTGTCCGTTTGCCGGTCGTGGGCGGCATGGTGCAAGTTGAATAACCGGAAATGGTTTCCCGCAGAACCTGAAGATGTTCGCGATTATCTTCTATATCTTCAGGCGCGCGGTCTGGCAGTAAAAACTATCCAGCAACATTTGGGCCAGCTAAACATGCTTCATCGTCGGTCCGGGCTGCCACGACCAAGTGACAGCAATGCTGTTTCACTGGTTATGCGGCGGATCCGAAAAGAAAACGTTGATGCCGGTGAACGTGCAAAACAGGCTCTAGCGTTCGAACGCACTGATTTCGACCAGGTTCGTTCACTCATGGAAAATAGCGATCGCTGCCAGGATATACGTAATCTGGCATTTCTGGGGATTGCTTATAACACCCTGTTACGTATAGCCGAAATTGCCAGGATCAGGGTTAAAGATATCTCACGTACTGACGGTGGGAGAATGTTAATCCATATTGGCAGAACGAAAACGCTGGTTAGCACCGCAGGTGTAGAGAAGGCACTTAGCCTGGGGGTAACTAAACTGGTCGAGCGATGGATTTCCGTCTCTGGTGTAGCTGATGATCCGAATAACTACCTGTTTTGCCGGGTCAGAAAAAATGGTGTTGCCGCGCCATCTGCCACCAGCCAGCTATCAACTCGCGCCCTGGAAGGGATTTTTGAAGCAACTCATCGATTGATTTACGGCGCTAAGGATGACTCTGGTCAGAGATACCTGGCCTGGTCTGGACACAGTGCCCGTGTCGGAGCCGCGCGAGATATGGCCCGCGCTGGAGTTTCAATACCGGAGATCATGCAAGCTGGTGGCTGGACCAATGTAAATATTGTCATGAACTATATCCGTAACCTGGATAGTGAAACAGGGGCAATGGTGCGCCTGCTGGAAGATGGCGATCTCGAGCCATCTGCTGGAGACATGAGAGCTGCCAACCTTTGGCCAAGCCCGCTCATGATCAAACGCTCTAAGAAGAACAGCCTGGCCTTGTCCCTGACGGCCGACCAGATGGTCAGTGCCTTGTTGGATGCTGAGCCCCCCATACTCTATTCCGAGTATGATCCTACCAGACCCTTCAGTGAAGCTTCGATGATGGGCTTACTGACCAACCTGGCAGACAGGGAGCTGGTTCACATGATCAACTGGGCGAAGAGGGTGCCAGGCTTTGTGGATTTGACCCTCCATGATCAGGTCCACCTTCTAGAATGTGCCTGGCTAGAGATCCTGATGATTGGTCTCGTCTGGCGCTCCATGGAGCACCCAGTGAAGCTACTGTTTGCTCCTAACTTGCTCTTGGACAGGAACCAGGGAAAATGTGTAGAGGGCATGGTGGAGATCTTCGACATGCTGCTGGCTACATCATCTCGGTTCCGCATGATGAATCTGCAGGGAGAGGAGTTTGTGTGCCTCAAATCTATTATTTTGCTTAATTCTGGAGTGTACACATTTCTGTCCAGCACCCTGAAGTCTCTGGAAGAGAAGGACCATATCCACCGAGTCCTGGACAAGATCACAGACACTTTGATCCACCTGATGGCCAAGGCAGGCCTGACCCTGCAGCAGCAGCACCAGCGGCTGGCCCAGCTCCTCCTCATCCTCTCCCACATCAGGCACATGAGTAACAAAGGCATGGAGCATCTGTACAGCATGAAGTGCAAGAACGTGGTGCCCCTCTATGACCTGCTGCTGGAGGCGGCGGACGCCCACCGCCTACATGCGCCCACTAGCCGTGGAGGGGCATCCGTGGAGGAGACGGACCAAAGCCACTTGGCCACTGCGGGCTCTACTTCATCGCATTCCTTGCAAAAGTATTACATCACGGGGGAGGCAGAGGGTTTCCCTGCCACAGCTTGATGAAGATCTGAGCTCCCTGGCGGAATTCCGAAGTTCCTATTCTCTAGAAAGTATAGGAACTTC-ATCAGTCAGGTACATAGGCGCGCCGGTCCCTGGGGCAACTTGCCCCATCCAGTGGCCCAGCCAACCCTTCCCAAGCCCTGAGTCTCCTCACCTCAGTCCCCTTATCCCCCTGGGGTTGCAGGAGACCAAGGGAAAAAAACCCTTTCCCTTCCTACAGGAAACCCTCTGGAGACGGAAAACCAGTGTGCCATCTACCCATGCTTAGTGACCCAGAGTGGCCCCTTGCCTTCCCCTCTTTCTTCAGAGGGGTTCCTAGGCATCCTGCAGTGACCTCCAGCTCACATCCACCTTCTCTGTATCGTGGCCTCGGTCCTGTCTCAGTGCAGAGATTGAGGCTCAATTTGAACCAAGCACCTAGTTATCAGAAGAAAATGGTGCCAAAGACAAGGCCCTGGAGTCCTTGACCTCTGAGTCGTGGGTGCCCTGGCTATGGGTGTAGGTGGAGCCCATGGGTGTCCTCAGTCACAGAGCTGGGAGCTCTCTCTCGCTCGCTTGGCATCAGGACTGCAGCCTCTTTCACTGGACACTGAGATGAGTCCCCAGGGTGTTCCCAGGGGAGAAAGCAGGTAACATCCCAGCTTTACCTAGGAATCCAGAGGACTTTAGGACTGTCCCTATGCACCCTGCAGGGCATCAGGAGACCAGGAAGGGATTCTAGCAGAGGGTAGGGGGCACAGAGGCAGAGCTGATTGCCCATGGGCTATCCCAGAATGCCTGGTCCTGAATCTAGCATCAGGAGGTGCAGGATTCCTAGGCTGCAATCTGACAGAGGCTTGCCCACTGTGTCAGGCCTGGGCAGCCCACAGAACCTGTCACTCTCCTCAATTGGTAGGAGAAGAGGTCTTGAGGTGACAGGAGGCAGAAGGCAGGCTCAGACAGTCAGAGAGCACCAAGCTTTCAAGTCCGCACCCCTGGGGTTCGGCATACCATCTTGCTGGCAGCTGGAAACCTGGTTCCCTGAAAGGGGGCCTCCATCCTCCAGAATGTAAGGCTCTTGATGCCACCGGATGCAGAGAGCCTTCTCGGGCCAGACAAAATTGCTGCTCCACCCCAGAGAAGATGTTCCAGCCTTCTTGGCATCTTAGAGGAAGGCCATGTCGCTGTCCTTTTCAGAGTAGCATATTTTTCAGTGATGGCTGCTCAGTCAGGAGGCTTCTGTCGCCTTACAAAGCACAGTGCGCTCTGGGCACTGTTTCTAAGCCACCCCATCCCACCCCCACCCCCGCCACCCCGGGGACAGAGGAAGATGCTAAAAGTCCCAGCAAAGAGGACAAAGCACCTTTCTTAAGCACTCCAGAGTCTTCCTTGTACCCCGCCCTCTCTTAGAGCTGGGTCTTTTGAGGGAAACGGATTGCTGAGCCCTCCCCCCCCCCCAATCCTCTCCTCTGTGGAGCTGTTTATCATCCTCTATTTATCAAAATCGCATCCATCTTTACCCTCTCCTTCGCTATAGCCTACTTCTGGATCACCCTCATCCAGTGCTGGTACCCCACAGCACTAAATCCAGGAACCCTGGGCTTGATCACCTGTTGCCACCTGTACACATGAAACCACCTGCTGGCCCGGCCCATGTCTCCTGCCCTCAGCCAGCAAGACATTCCTAGAGAGAGGAACTATGGGCTTAAAAGCCCCAACTGACTTCCTTTTGCCTGGGGACCTGAACCGACAAGACACCAGGGACACTTGTCTACATGAACATGTGACCAATGTACACCGATTTCTCATCTCTAGACCTATTATCTGAAGCCTGTCCCGGGCCATGACTAGAATGGCTTGTATCTGTGGTTTAGAGAAGTCTAATAATAACTGAGGGCAAACTGACTCTCTGGTAGCATGGAGCACCAGGCGGATGGAGCTCACCAGCTCTGTCCAGGTTTCAAAGGAGGAGACTGTTGGGCTCTTCAAGGTCTGGACAAGAGGAAAGCCACATTGCCCCCTTGGGAACCCAGGTTCTCCTTTTGAACTTCTCACAGCTGCAAGCACCCCTTTCAAAGACCAAATGCATCCTCCTCCACATTCCTTGCTCCCTGGAGGCCTGGCTCTGGATACACCTGAGTCTTCGTTCACCTACTACACTTTAGGAGCAGGAACTTCAAGCAGGTGACATCCACAGGGCCCAGTCCCAGCCAAGGGAGCAACATTCCAACGCTTGGACCAATCATAATGATCTGCCCGTGAGGGTAACCGCAACTAGAGACCTGCTTGGGAGAAAACAAAATGACTTCTCATTCCATGCCATGCCTCTGAATGCTCCCCCAAGCTGCCATCTTGGTATAAAATGGGACTTGTGTTGTGGGGAACCCCTTGACCCCAACAGGTTTTCCCAACTGTCTCATGCTTTTGTGAATCTGTCTGCTTTGATCTGTAAAACTCAGCCTTGTTTGGGCAGCTTGTAATTTCAACAGTGAGGCGACATCGATTAGATGAGAGGCACCAGGCCTCTCCGCCGCCGTCCCTCTGTGGCCGTCCCTCTGGGGTTGAGCAGAACCTAGAAGAAGGCCGATTTCCAGTGGCCAGACTGGACCAGAAACAGCCCCCACCCCAATCCCTGTAAATAGAGTCAATAGCAAAATAAGAGGGGCGCCCTCCATGTCACCTCAAGTAGCTACTGGTTCTTCTGTGGAGGCCCCTCTGAACTCATTGTCTGGTAGTTGAAAATGTGATGTTGTGCTGTTTGTTTATAGAACATTGGCTTTTTATATATAAATCTATATACTTAAAAACAAAAACAAAAACAAAAAAAATGGAAAGAAACCCCCACAGTTTGTGTCGTGCTAGTGCCGGGGACCATCTGTAGGGATCTCATTGTGTGTTTGGATTATGTATGCCAGCCTTTCTCGGGTTCCTCCTTCTTTTTTTTTTTCTTTGTCATTTCCCTACTCCCGATTTGGTTTTTCTCTCTCGCTCTGAAAATGAAAGAGATGGCTTGAAGAGCCAAGCTCCTACTCCTTTACTTTATAGGCGAATTAACTTTGCCCTTCTTTTAGGGGACCAGGAACAAGCACAGGAAACAGAAATTAGGAAGCAAGAGGGTAGGAGAGAAGAGTGAGGGGAGGAGCTAGCCAGGGAGACAGGAGAGCAGGGTGGTTTCGCAGCCAGATGATGGCTGTGGCCCAGGACGGAGCCTGGATTAGCATGCTCCACCAGAGGCTGGTATTGTTTTGCATGGAAATGGTAGAATGTTCTCAGACATGGAGCTAAGCCTTGCTGAGGGACTCCACAGTAGCCAACTAAGAAGACTCCTTTAAGAGGCTCCAATTCTCCCCAGGATCTTGGAAATGTCTCTCCCTGCAGGATGCTTCTCTTGGGTACACAAGCATCCTCGCCTTCTATAAAAGCACCAAGCCAAGACCAAATGGAGGGTATGTGGGCTGCCGTGTTGAAATGTGTCAAGATACCTGTGCATTTAATGTCTCTTCTGTCTGTACGGACATGGGCAGCCCCCTGCTGGGAAAAGGCCACCCTGTGCTGACGAACATTCCTTCTTCCCAGTGACTGCAACATTGTTGCTTCAGTGTTTGGCTCAGCCATTTCTCCTCCACGCAAAGTGAGTGTGTGTGCATGAGGGGTGTGAGTGGGATGTGTGCGTGTGGGCGGGCGTCTAAATATAACCGCACCATAATTTATTCAGCTATATGGAGTAGTAGACTAGAAAGACAAACTGGTATTTGCTTTAACTACCTGCTGCTTGTAAGCGCTGCCTCTGTAACTCAGGCGTAACTGTTATACATTAAAAGAAATTAAAAACATCATGAAGATTCTGCATCCTCAGTTGACTGCCATCCTGTGGCTGGGGACCAGGGTGTCTGGGAACCACACGATGGGTTGGTGTGGGAGTGTTTCTAGTGTACCCTGACCCAGCTTTCTTGTACAAAGTGGT"

4.`tail genes\genes.gtf > temp_genes.gtf`. 10X website has the details on the columns. Use a text editor to add each gene on a new line. 
  
`eGFP	annotation	exon	1	715	.	+	.	gene_id "eGFP"; transcript_id "eGFP"; gene_name "eGFP"; gene_biotype "protein_coding";`
`tdTWPRE	annotation	exon	1	2111	.	+	.	gene_id "tdTWPRE"; transcript_id "tdTWPRE"; gene_name "tdTWPRE"; gene_biotype "protein_coding";`
`ICET2	annotation	exon	1	6525	.	+	.	gene_id "ICET2"; transcript_id "ICET2"; gene_name "ICET2"; gene_biotype "protein_coding";`


cat GFP.fa | grep -v "^>" | tr -d "\n" | wc -c

5. Append the `GTF` files with `cat refdata-cellranger-mm10-3.0.0/genes/genes.gtf temp_genes.gtf > new_genes.gtf`.

6. Append the `fasta` files with `cat refdata-cellranger-mm10-3.0.0/fasta/genome.fa temp_genome.fa > new_genome.fa`.

7.You can verify this using `tail new_genes.gtf` or `grep -n eGFP new_genes.gtf`.

## Make reference file 

8.This step depends on your environment. Run the `cellranger mkref` using the following configuration. This will create a folder with the new 10xreference genome in the folder `10x_Fl`. 

```
#!/bin/bash -l
#PBS -N NewRef_for_10X
#PBS -l nodes=1:ppn=24,walltime=00:40:00,mem=60gb
#PBS -m abe
#PBS -M verma014@umn.edu

module load umgc
module load cellranger

cd .../reference/
cellranger mkref --genome=10x_Fl --fasta=new_genome.fa --genes=new_genes.gtf --nthreads=24 --memgb=60

```
9.You can verify the your new genes using the step above in the `genome.fa`, `genome.fa.fai` and `genes.gtf`.

## Count 

10.Use `cellranger count` command to map the reads. This is a very memory intensive process so plan ahead. 

```
#!/bin/bash -l
#PBS -l walltime=07:00:00,nodes=1:ppn=24,mem=250gb
#PBS -N Custom_genome_10x_count
#PBS -m abe
#PBS -M verma014@umn.edu

cd $PBS_O_WORKDIR
# Should be run with "qsub -q ram256g" with mesabi. No need to use it for Mangi. Normally takes about 5 hrs.
set -v
ulimit -n
ulimit -n 10000

module load umgc
module load cellranger

cellranger count --id=D0_Fl_genome --fastqs=./fastqfoldername --transcriptome=./10x_Fl --sample=Asakura_Project_016 --expect-cells=3000 --localcore=24 --localmem=200 --chemistry=auto
```
11.You can briefly view the genes with in the Loupe browser in the feature section. 

## Troubleshoot 

12.If you would like to troubleshoot issues with alignment, you can use IGV. Go to `Genomes>Create .genomes File...`. Get the FASTA file from `10x_Fl\fasta\genome.fa` and the `gene file` from `10x_Fl\genes\genes.gtf`. You will have to add a unique identifies and description. 

13.Open up the `BAM` file in IGV with the new genome from the cellranger count output directory in `dir\outs\possorted_genome_bam.bam.

14.Browse for the gene of interst and take a look at the reads that are mapped to each. Since you are using cDNA, there should be basically no spliced reads there.

15. Alternatively you could also look for a specific sequence in the bam file using samtools.
```
'samtools view `file.bam` | grep **YOUR_SEQUENCE** | awk '{print $3 "\t" $4 "\t" $4+length($10)-1}' > sequence.tab'
```
For example, for GFP:
```
'samtools view `file.bam` | grep CAGTCCGCCCTGAGCAAAGACCCCAACGAGAAGCGCGATCACATGGTCCTG
CTGGAGTTCGTGACCGCCGCCGGGATCACTCTCGGCATGGACGAGCTG | awk '{print $3 "\t" $4 "\t" $4+length($10)-1}' > GFPsequence.tab'
```
## Verify reads
The BAM files that comes out from the cellranger count includes all aliged reads including low quality reads and reads that are mapped to multiple genes. These are not used for UMI counts. The requirements are listed (here)[https://kb.10xgenomics.com/hc/en-us/articles/115003710383-Which-reads-are-considered-for-UMI-counting-by-Cell-Ranger-]

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(DropletUtils)
```
```{r}
mol.info.file <- "D:/Informatics/MVSingleCell/D3GTWC9/molecule_info.h5"
molinfo <- read10xMolInfo(mol.info.file)
```

You can pull out the reads that were counted for UMI counts from this file.
You will used SAM tools to extract those reads. Information for using 'samtools' for this can be found (here)[https://kb.10xgenomics.com/hc/en-us/articles/360007068611-How-do-I-get-the-read-counts-for-each-barcode-].
The instruction for getting the BAM file for specific barcodes can be found (here)[https://kb.10xgenomics.com/hc/en-us/articles/360022448251-Is-there-way-to-filter-the-BAM-file-produced-by-10x-pipelines-with-a-list-of-barcodes-]


```{r}

```
This will take as much time as it takes to read the whole sam file so be patient. 
```
module load samtools
samtools view file.bam | grep "MM:i:1" | cat header.sam - | samtools view -Sb - > UMIreads.bam
samtools sort UMIreads.bam -o UMIreads.sorted.bam
samtools index UMIreads.sorted.bam
```
Examine mapped reads using IGV: 
Open these up in IGV to find which reads were used for UMI counting. 
Alternatively, you can just group and color the MM tag in IGV. 

## Sanity check 
One of the things that surpised me is how well the alignment actually works.For example, this is the alighment between GFP and TdTomato. While the small sequences have good alignment, you still get ~90 bp reads with the 10x data after barcodes. In my spot checking, I could not find any reads that I could say were definate misalignments. This makes sense as reads that are mapped to multiple genes are thrown out and not accounted in the UMI count. This gives me confidence that the reads are not simply misread. 
In addition, having the WPRE segment in the TdTomatoWPRE is helpful. Since 10x is 3' biased, most of the reads are on the WPRE and not the tdTomato. I am hesitent to map just the WPRE and not the whole tdTomatoWPRE because I suspect with out it, there might be some reads that would be between tdTomato and GFP, they will no longer get thrown out. 

```
eGFP         GTGAGCAAGGGCGAGGAGCTGTTCACCGGGGTGGTGCCCATCCTGGTCGAGCTGGACGGC	60
dTomato      gtgagcaagggcgaggaggtcatcaaagagttcatgcgcttcaaggtgcgcatggagggc	60
             ****************** *  ***  * * *  *** * **  ***     **** ***

eGFP         GACGTAAACGGCCACAAGTTCAGCGTGTCCGGCGAGGGCGAGGGCGATGCCACCTACGGC	120
dTomato      tccatgaacggccacgagttcgagatcgagggcgagggcgagggccgcccctacgagggc	120
               * * ********* *****    *    ***************    **  * * ***

eGFP         AAGCTGACCCTGAAGTTCATCTGCACCAC---CGGCAAGCTGCCCGTGCCCTGGCCCACC	177
dTomato      acccagaccgccaagctgaaggtgaccaagggcggccccctgcccttcgcctgggacatc	180
             *  * ****   *** * *     ****    ****   ****** *  *****  ** *

eGFP         CTCGTGACCACCCTGACCTACGGCGTGCAGTGCTTCAGCCGCTACCCCGACCACATGAAG	237
dTomato      ctgtccccccagttcatgtacggctccaaggcgtacgtgaagcaccccgccgacatc---	237
             **     **    * *  ******    **   * *       ****** * ****    

eGFP         CAGCACGACTTCTTCAAGTCCGCCATGCCCGAAGGCTACGTCCAGGAGCGCACCATCTTC	297
dTomato      ---cccgattacaagaagctgtccttccccgagggcttcaagtgggagcgcgtgatgaac	294
                * *** * *   ***    ** * ***** **** *     *******   **   *

eGFP         TTCAAGGACGACGGCAACTACAAGACCCGCGCCGAGGTG---AAGTTCGAGGGCGACACC	354
dTomato      ttcgaggacggcggtctgg---tgaccgtgacccaggactcctccctgcaggacggcacg	351
             *** ****** ***         ****    ** ***         *  *** ** *** 

eGFP         CTGGTGAACCGCATCGAGCTGAAGGGCATCGACTTCAAGGAGGACGGCAACATCCTGGGG	414
dTomato      ctgatctacaaggtgaagatgcgcggcaccaacttcccccccgacggccccgtaatgcag	411
             *** *  **    *  ** **   **** * *****      ******  * *  **  *

eGFP         CACAAGC---TGGAGTACAACTACAACAGCCACAACGTCTATATCATGGCCGACAAGCAG	471
dTomato      aagaagaccatgggctgggaggcctccaccg--agcgcctgtacccccgc----------	459
              * ***    ***  *   *   *  ** *   * ** ** ** *   **          

eGFP         AAGAACGGCATCAAGGTGAACTTCAAGATCCGCCACAACATCGAGGACGGCAGCGTGCAG	531
dTomato      ---gacgg---cgtgctgaagggcgagatccaccaggccctg---------aagctgaag	504
                 ****   *  * ****   * ****** ***   * *          *   ** **

eGFP         CTCGCCGACCACTACCAG---------CAGAACACCCCCATCGGCGACGGCCCCGTGCTG	582
dTomato      gacggcggccactacctggtggagttcaagaccatctacatggccaagaagcccgtgcaa	564
               ** ** ******** *          *** ** *  *** * * *    *******  

eGFP         CTGCCCGACAACCACTACCTGAGCACCCAGTCCGCCCTGAGCAAAGACCCCAACGAGAAG	642
dTomato      ctgcccggctactactacgtggacaccaagctggacatcacct---cccacaacgagg--	619
             ******* * ** ***** **  **** **   * * * * *     ** *******   

eGFP         CGCGATCACATGGTCCTGCTGGAGTTCGTGACC-------GCCGCCGG-----GATCACT	690
dTomato      ----actacaccatcgtggaacagtacgagcgctccgagggccgccaccacctgttcctg	675
                 *  ***   ** **    *** ** *  *       ******       * **   

eGFP         CTCGGCATGGACGAGCTGTACAAG	714
dTomato      tacggcatggacgagctgtacaag	699
               **********************
```

  

  ls -R | grep ":$" | sed -e 's/:$//' -e 's/[^-][^\/]*\//--/g' -e 's/^/   /' -e 's/-/|/'

```
This is the last sequence that comes up when you have the tdTomato_WPRE
Query  5942  CTTCGCCCTCAGACGAGTCGGATCTCCCTTTGGGCCGCCTCCCCGCATCGATACCGTCGA  6001
             ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
Sbjct  1     CTTCGCCCTCAGACGAGTCGGATCTCCCTTTGGGCCGCCTCCCCGCATCGATACCGTCGA  60

Query  6002  CCTCGACCTCGACTGTGCCTTCTAGTTGCCAGCCATCT  6039
             ||||||||||||||||||||||||||||||||||||||
Sbjct  61    CCTCGACCTCGACTGTGCCTTCTAGTTGCCAGCCATCT  98
```
```
For GFP
Query  6807  CTGAGCAAAGACCCCAACGAGAAGCGCGATCACATGGTCCTGCTGGAGTTCGTGACCGCC  6866
             ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
Sbjct  1     CTGAGCAAAGACCCCAACGAGAAGCGCGATCACATGGTCCTGCTGGAGTTCGTGACCGCC  60

Query  6867  GCCGGGATCACTCTCGGCATGGACGAGCTGTACAAGTA  6904
             ||||||||||||||||||||||||||||||||||||||
Sbjct  61    GCCGGGATCACTCTCGGCATGGACGAGCTGTACAAGTA  98
```
This is the eGFP with the last read overhang in the bold
```
>eGFP
GTGAGCAAGGGCGAGGAGCTGTTCACCGGGGTGGTGCCCATCCTGGTCGAGCTGGACGGCGACGTAAACGGCCACAAGTT
CAGCGTGTCCGGCGAGGGCGAGGGCGATGCCACCTACGGCAAGCTGACCCTGAAGTTCATCTGCACCACCGGCAAGCTGC
CCGTGCCCTGGCCCACCCTCGTGACCACCCTGACCTACGGCGTGCAGTGCTTCAGCCGCTACCCCGACCACATGAAGCAG
CACGACTTCTTCAAGTCCGCCATGCCCGAAGGCTACGTCCAGGAGCGCACCATCTTCTTCAAGGACGACGGCAACTACAA
GACCCGCGCCGAGGTGAAGTTCGAGGGCGACACCCTGGTGAACCGCATCGAGCTGAAGGGCATCGACTTCAAGGAGGACG
GCAACATCCTGGGGCACAAGCTGGAGTACAACTACAACAGCCACAACGTCTATATCATGGCCGACAAGCAGAAGAACGGC
ATCAAGGTGAACTTCAAGATCCGCCACAACATCGAGGACGGCAGCGTGCAGCTCGCCGACCACTACCAGCAGAACACCCC
CATCGGCGACGGCCCCGTGCTGCTGCCCGACAACCACTACCTGAGCACCCAGTCCGCCCTGAGCAAAGACCCCAACGAGA
AGCGCGATCACATGGTCCTGCTGGAGTTCGTGACCGCCGCCGGGATCACTCTCGGCATGGACGAGCTGTACAAG**TAAAGCGGCCGGCCGCGACTCTAGATCATAAT**
```


-----------------------------------------
## Change log 
Both eGFP and tdtWPRE work in the `10x_Fl` transcriptome. 
tdtWPRE works in `10x_Fl`
GTWC4. Added space after each. Did not work.
GTWC5. Addd Ires-CreERt2-Pax7(9i)Moved all transcripts to start at 1.All3 mapped.GFP without the overhang(ends in CAAG)
GTWC6. Added the overhangs from the last read for TdTWPRE and GFP. Those overhangs dont seem like they didnt come up in the genome file. They are present in the 
GTWC7. Overhangs worked after changing to FASTA format. There is additional overhang for GFP(CAGCCATACCACATTTGTAGAGGTTTTACT) and tdtwpre(TGCCCCTCCCCCGTGCCTTCCTTGACCCTGGA). Made the artificial chromosome with SV40(eGFP) and bGH(tdt) polyA sequence.
GTWC8. Complete eGFPSV40 and tdTWPREBgH. There are a bunch of /n in the tdtomato.Tdt and cre work but GFP has 0 reads.
GTWC9.tdt to end of attb site and removed ns. Labeling changes.
GTWC10. In the off chance the eGFP and Tdtomato, just map to SV40 and WPRE.

In the end, I would like to chop off the end of Pax7 exon9 with the 3' UTR which would map to the CRE. Did not work & not worth the effort anymore. 

```
SV40_polyA	annotation	exon	101	328	.	+	.	gene_id "SV40_polyA"; transcript_id "SV40_polyA"; gene_name "SV40_polyA"; gene_biotype "protein_coding";
tdTWPREbGH	annotation	exon	1	2360	.	+	.	gene_id "tdTWPREbGH"; transcript_id "tdTWPREbGH"; gene_name "tdTWPREbGH"; gene_biotype "protein_coding";
ICET2Pax7	annotation	exon	1	6523	.	+	.	gene_id "ICET2Pax7"; transcript_id "ICET2Pax7"; gene_name "ICET2Pax7"; gene_biotype "protein_coding";

                                          